shared/src/metabase/shared/models/visualization_settings.cljc

```diff
@@ -0,0 +1,410 @@
+      "name" {::column-name (first params)})))
+
+(defn- with-col-settings [settings]
+  (if (contains? settings ::column-settings)
+    settings
+    (assoc settings ::column-settings {})))
+
+(defn crossfilter-click-action
+  "Creates a crossfilter click action with the given `param-mapping`, in the normalized form."
+  {:added "0.40.0"}
+  [param-mapping]
+  {::click-behavior-type ::cross-filter
+   ::parameter-mapping   param-mapping})
+
+(defn url-click-action
+  "Creates a URL click action linking to a `url-template`, in the normalized form."
+  {:added "0.40.0"}
+  [url-template]
+  {::click-behavior-type ::link
+   ::link-type           ::url
+   ::link-template       url-template})
+
+(defn entity-click-action
+  "Creates a click action linking to an entity having `entity-type` with ID `entity-id`, in the normalized form.
+  `parameter-mapping` is an optional argument."
+  {:added "0.40.0"}
+  [entity-type entity-id & [parameter-mapping]]
+  (cond-> {::click-behavior-type ::link
+           ::link-type           entity-type
+           ::link-target-id      entity-id}
+          (some? parameter-mapping) (assoc ::parameter-mapping parameter-mapping)))
+
+(defn with-click-action
+  "Creates a click action from a given `from-field-id` Field identifier to the given `to-entity-type` having ID
+  `to-entity-id`, and adds it to the given `settings`. This happens in the normalized form, and hence this should be
+  passed the output of another fn (including, currently, `visualization-settings`). If the given `from-field-id`
+  already has a click action, it will be replaced."
+  {:added "0.40.0"}
+  [settings col-key action]
+  (-> settings
+      with-col-settings
+      (update-in [::column-settings] #(assoc % col-key {::click-behavior action}))))
+
+(defn with-entity-click-action
+  "Creates a click action from a given `from-field-id` Field identifier to the given `to-entity-type` having ID
+  `to-entity-id`. This happens in the normalized form, and hence this should be passed the output of another fn
+  (including, currently, `visualization-settings`). If the given `from-field-id` already has a click action, it will
+  be replaced."
+  {:added "0.40.0"}
+  [settings from-field-id to-entity-type to-entity-id & [parameter-mapping]]
+  (with-click-action settings (column-ref-for-id from-field-id) (entity-click-action
+                                                                 to-entity-type
+                                                                 to-entity-id
+                                                                 parameter-mapping)))
+
+(defn fk-parameter-mapping
+  "Creates a parameter mapping for `source-col-name` (`source-field-id`) to `target-field-id` in normalized form."
+  {:added "0.40.0"}
+  [source-col-name source-field-id target-field-id]
+  (let [id         [:dimension [:fk-> [:field-id source-field-id] [:field-id target-field-id]]]
```

Comment:
```clj
[:fk-> [:field-id source-field-id] [:field-id target-field-id]]
```

is legacy MBQL syntax, don't use it please 😬 

You want 

```clj
[:field target-field-id {:source-field source-field-id}]
```

instead

Reply:
Fixed.